### PR TITLE
#1587 Fix --null-value option not handling spaces

### LINKED ISSUE #1587: --null values should handle empty spaces
## Describe the bug
--null values should handle empty spaces

### PR DESCRIPTION
**Pull request by Zejnilovic**
Closes #1587 

RN: --null-value in helper scripts now encapsulates empty spaces in `'`

### CODE REVIEW

File: scripts/bash/run_enceladus.sh [hunk 1]
```diff
@@ -197,7 +197,11 @@ case $key in
     shift 2 # past argument and value
     ;;
   --null-value)
-    NULL_VALUE="$2"
+    if [[ -z "${2// }" ]]; then
```
Comment (benedeki) [inline scripts/bash/run_enceladus.sh:200]: Sorry, how this exactly works?
Not sure I understand the slashes there. And it's different from `--delimiter`. Shouldn't it be the same?

Comment (Zejnilovic) [reply scripts/bash/run_enceladus.sh:200]: The delimiter is a char, null-value is a string. So delimiter caters for a single char being a white space. This needs to cater for a whole string being a whitespace.

Comment (Zejnilovic) [reply scripts/bash/run_enceladus.sh:200]: As to why and what - https://www.gnu.org/savannah-checkouts/gnu/bash/manual/bash.html#Shell-Parameter-Expansion

**Comment by Zejnilovic** [comment]
Tip for testing scripts. Run in `--dry-run` mode.

**Review by benedeki — CHANGES_REQUESTED** [review]
`run_standardization.sh --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value 'hey you'`
`run_standardization.sh --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value " hey "'`
etc, doesn't produce the desired result. Separates the two words and strips the trailing white spaces.

**Review by benedeki — DISMISSED** [review]
After discussion, we agreed that these are to be expected, can be circumvented by using `\"` and generally good enough.
- [x] code reviewed
- [x] pulled
- [x] built
- [x] run

**Comment by dk1844** [comment]
Having test-ran on Windows, I think that the null-value empty spaces handling needs more work:

## 'middle space'
```
run_standardization.cmd --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value 'hey you'
```
result not correct IMHO:
```
Command line:
MY_SPARK_PATH/bin/spark-submit --master local --deploy-mode "client"  --conf spark.logConf=true --conf spark.sql.adaptive.shuffle.targetPostShuffleInputSize=134217728 --conf "spark.driver.extr
aJavaOptions=-Dstandardized.hdfs.path="/bigdata/std/std-{0}-{1}-{2}-{3}" -Dspline.mongodb.url=mongodb://localhost:27017 -Dspline.mongodb.name=spline -Dhdp.version=2.7.1   " --class za.co.absa.enceladus.standardization.StandardizationJo
b MY_ENCELADUS_PATH\spark-jobs-2.13.0-SNAPSHOT.jar --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --null-value 'hey
```

## "middle space"
```
run_standardization.cmd --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value "hey you"
```
result not correct IMHO:
```
you""=="" was unexpected at this time.
```

## "  leading and trailing  "
```
run_standardization.cmd --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value "  hey you  "
```
result not correct IMHO:
```
hey was unexpected at this time.
```

## "  leading and trailing  "
```
run_standardization.cmd --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --null-value '  hey you  '
```
result not correct IMHO:
```
Command line:
MY_SPARK_PATH/bin/spark-submit --master local --deploy-mode "client"  --conf spark.logConf=true --conf spark.sql.adaptive.shuffle.targetPostShuffleInputSize=134217728 --conf "spark.driver.extr
aJavaOptions=-Dstandardized.hdfs.path="/bigdata/std/std-{0}-{1}-{2}-{3}" -Dspline.mongodb.url=mongodb://localhost:27017 -Dspline.mongodb.name=spline -Dhdp.version=2.7.1   " --class za.co.absa.enceladus.standardization.StandardizationJo
b MY_ENCELADUS_PATH\spark-jobs-2.13.0-SNAPSHOT.jar --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --null-value '
```


Or have I made a mistake while testing it somehow?

**Comment by Zejnilovic** [comment]
@dk1844, this is the same question as asked by @benedeki that ended with

> After discussion, we agreed that these are to be expected, can be circumvented by using \ and generally good enough.

Covering for each of the complex cases would require too much bash logic. If people have a complex case then they should use `"\" alfa beta \""`. This fix is just to mimic the delimiter options.

**Comment by dk1844** [comment]
Test-ran bash (on Cygwin), LGTM.
```./run_standardization.sh --dataset-name Foo --dataset-version 5 --report-date 2018-07-16 --report-version 1 --raw-format csv --dry-run --menas-credentials-file myfile1 --null-value "\"  hey you  \""``` -> ```...  --null-value "  hey you  "``` 👍  (conformance = same)

**Review by AdrianOlosutean — APPROVED** [review]

**Review by dk1844 — APPROVED** [review]

**Pull request merged**